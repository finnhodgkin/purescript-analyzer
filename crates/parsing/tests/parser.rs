macro_rules! parser_tests {
    ($($name:ident => $source:expr),+ $(,)?) => {
        $(
            #[test]
            fn $name() {
                let lexed = lexing::lex($source);
                let tokens = lexing::layout(&lexed);
                let (node, errors) = parsing::parse(&lexed, &tokens);
                insta::assert_debug_snapshot!((node, errors));
            }
        )+
    };
}

parser_tests!(
    comment_pre_post => include_str!("parser/CommentPrePost.purs"),
    lexer_error_inserted => include_str!("parser/LexerErrorInserted.purs"),
    module_export => include_str!("parser/ModuleExport.purs"),
    module_export_class => include_str!("parser/ModuleExportClass.purs"),
    module_export_class_error_end => include_str!("parser/ModuleExportClassErrorEnd.purs"),
    module_export_comma_error => include_str!("parser/ModuleExportCommaError.purs"),
    module_export_data => include_str!("parser/ModuleExportData.purs"),
    module_export_data_comma_error => include_str!("parser/ModuleExportDataCommaError.purs"),
    module_export_data_item_after_double_period => include_str!("parser/ModuleExportDataItemAfterDoublePeriod.purs"),
    module_export_data_item_before_double_period => include_str!("parser/ModuleExportDataItemBeforeDoublePeriod.purs"),
    module_export_data_item_invalid => include_str!("parser/ModuleExportDataItemInvalid.purs"),
    module_export_empty => include_str!("parser/ModuleExportEmpty.purs"),
    module_export_invalid => include_str!("parser/ModuleExportInvalid.purs"),
    module_export_type_operator => include_str!("parser/ModuleExportTypeOperator.purs"),
    module_export_type_operator_error => include_str!("parser/ModuleExportTypeOperatorError.purs"),
    module_export_type_operator_error_end => include_str!("parser/ModuleExportTypeOperatorErrorEnd.purs"),
    module_header => include_str!("parser/ModuleHeader.purs"),
    module_header_prefixed => include_str!("parser/ModuleHeaderPrefixed.purs"),
);
