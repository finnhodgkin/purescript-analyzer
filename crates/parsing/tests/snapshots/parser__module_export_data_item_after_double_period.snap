---
source: crates/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..62
      ModuleHeader@0..62
        MODULE@0..6 "module"
        Comment@6..7
          WHITESPACE@6..7 " "
        ModuleName@7..45
          UPPER@7..44 "ModuleExportDataItemA ..."
          Comment@44..45
            WHITESPACE@44..45 " "
        ModuleExportList@45..62
          LEFT_PARENTHESIS@45..46 "("
          ModuleExportType@46..53
            UPPER@46..50 "List"
            ModuleExportTypeItemsAll@50..53
              LEFT_PARENTHESIS@50..51 "("
              DOUBLE_PERIOD@51..53 ".."
              ERROR@53..53 ""
          COMMA@53..54 ","
          Comment@54..55
            WHITESPACE@54..55 " "
          ModuleExportType@55..61
            UPPER@55..61 "Period"
            ModuleExportTypeItemsList@61..61
          RIGHT_PARENTHESIS@61..62 ")"
        ERROR@62..62 ""
      ERROR@62..62 ""
      ERROR@62..62 ""
      ERROR@62..62 ""
    ,
    [
        ParseError {
            position: Position {
                offset: 53,
                line: 1,
                column: 54,
            },
            message: "expected RIGHT_PARENTHESIS",
        },
        ParseError {
            position: Position {
                offset: 62,
                line: 1,
                column: 63,
            },
            message: "expected WHERE",
        },
        ParseError {
            position: Position {
                offset: 62,
                line: 1,
                column: 63,
            },
            message: "expected LAYOUT_START",
        },
        ParseError {
            position: Position {
                offset: 62,
                line: 1,
                column: 63,
            },
            message: "expected LAYOUT_END",
        },
        ParseError {
            position: Position {
                offset: 62,
                line: 1,
                column: 63,
            },
            message: "expected END_OF_FILE",
        },
    ],
)
